src/vcpkg/registries.cpp

```diff
@@ -799,31 +805,49 @@ namespace
         return res;
     }
 
-    ExpectedL<Version> BuiltinGitRegistry::get_baseline_version(StringView port_name) const
+    ExpectedL<Optional<Version>> BuiltinGitRegistry::get_baseline_version(StringView port_name) const
     {
-        const auto& baseline = m_baseline.get([this]() -> Baseline {
+        const auto& maybe_baseline = m_baseline.get([this]() -> ExpectedL<Baseline> {
             auto maybe_path = git_checkout_baseline(m_paths, m_baseline_identifier);
-            if (!maybe_path)
+            auto path = maybe_path.get();
+            if (!path)
             {
-                msg::println(Color::error, LocalizedString::from_raw(maybe_path.error()));
-                msg::println(Color::error, LocalizedString::from_raw(m_paths.get_current_git_sha_baseline_message()));
-                Checks::exit_fail(VCPKG_LINE_INFO);
+                return std::move(maybe_path)
+                    .error()
+                    .append_raw('\n')
+                    .append_raw(m_paths.get_current_git_sha_baseline_message());
             }
-            auto b = load_baseline_versions(m_paths.get_filesystem(), *maybe_path.get()).value_or_exit(VCPKG_LINE_INFO);
-            if (auto p = b.get())
+
+            auto maybe_maybe_baseline = load_baseline_versions(m_paths.get_filesystem(), *path);
```

Comment:
Are there any users that care about the `ExpectedL<>(nullopt)` case? This seems ripe for simplification to just `ExpectedL<Baseline>`.

I think consumers like `BuiltinGitRegistry` should be adding their context information (`m_baseline_identifier`) always -- why should "file not found" be special compared to "parse error"?

Reply:
`ExpectedL<>(nullopt)` means 'that port is not in the baseline': https://github.com/microsoft/vcpkg-tool/pull/1203/files/38f738d8100a9119d08bf640fca4d5dceaef15ee#diff-9439a305925c63597b93bb0d49db1794b5b6120f7ca5456d3e99b403c862dbf5R93-R96

For instance, `load_all_control_files` ignores ports not in the baseline, but emits an error if there are errors:

https://github.com/microsoft/vcpkg-tool/pull/1203/files/38f738d8100a9119d08bf640fca4d5dceaef15ee#diff-d4fe92b1ebc56fd05568a18c509672434c578cf4e147b2d5aad36226c026ff00R547-R549

Both 'file not found' and 'parse failure' are errors. But 'parse success and the port you asked for is not in there' is not an error.

Reply:
By my reading, this call is not "Get the baseline for a port" -- this is "Load the baseline file".

```cpp
ExpectedT<Optional<Baseline>> load_baseline_versions(fs, path);
```

Agreed that for `ExpectedT<Optional<Version>>`, `nullopt` means "not in baseline". But for this function (`load_baseline_versions`), as far as I can tell `nullopt` means "The baseline.json didn't exist at all" -- which _seems_ the same as a parse error.

Reply:
Ah, I see. Thanks GH for making a totally reasonable comment opaque.

It looks like this was here because some callers emitted different error messages for not exist vs. parse error. I think, in general, sometimes this can make sense; I certainly used it for `load_git_versions_file` in `x-ci-verify-versions` to do something different for 'the versions file is corrupted' vs. 'this port does not have a versions file yet'.

However, I can't think of obvious reasons that would want to happen for baseline things, so I've ripped this one out.